tests/features/async-components.spec.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  defineAsyncComponent,
+  defineComponent,
+  h,
+  ComponentPublicInstance
+} from 'vue'
+import flushPromises from 'flush-promises'
+
+import { mount } from '../../src'
+
+// AsyncComponents are documented here: https://github.com/vuejs/rfcs/blob/async-component/active-rfcs/0026-async-component-api.md
+
+describe('defineAsyncComponent', () => {
+  it('works with the basic usage', async () => {
+    const AsyncHello = defineAsyncComponent(() =>
+      import('../components/Hello.vue')
+    )
+    const Comp = defineComponent({
+      render() {
+        return h('div', [h(AsyncHello)])
+      }
+    })
+
+    const wrapper = mount(Comp)
+    await flushPromises()
+    expect(wrapper.html()).toContain('Hello world')
+  })
+
+  it('works with options usage', async (done) => {
+    const Async = defineAsyncComponent({
+      loader: () =>
+        new Promise<any>((res) => {
+          setTimeout(() => {
+            res({
+              template: '<div>Async Component</div>'
+            })
+          }, 75)
+        }),
+      loadingComponent: {
+        template: '<div>Loading Component</div>'
+      },
+      delay: 10
+    })
+
+    const Comp = defineComponent({
+      render() {
+        return h('div', [h(Async)])
+      }
+    })
+
+    const wrapper = mount(Comp)
+
+    setTimeout(() => {
+      expect(wrapper.html()).toContain('Loading Component')
+    }, 35)
+
+    setTimeout(() => {
+      expect(wrapper.html()).toContain('Async Component')
+      done()
+    }, 100)
```

Comment:
Why not use [jest.useFakeTimers](https://jestjs.io/docs/en/timer-mocks)? Mocking the `setTimeout` would make the test 10 times faster.
Something like this seems to work fine, but there may be something I'm not seeing:

```suggestion
    await jest.advanceTimersByTime(35)
    expect(wrapper.html()).toContain('Loading Component')

    jest.advanceTimersByTime(100)
    await flushPromises()
    expect(wrapper.html()).toContain('Async Component')
```

Reply:
I did not know jest had a fake timer, that is pretty cool.